### PR TITLE
feat: add key vault support in static provisioning scenario

### LINKED ISSUE #69: read SAS token from Azure KeyVault (static provisioning support)
**Describe the solution you'd like in detail**
Add ability to read SAS tokens that are mounted by Azure KeyVault flexvolume to some path instead of K8S secrets

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add key vault support in static provisioning scenario

**Which issue(s) this PR fixes**:

Fixes #69 

**Special notes for your reviewer**:
Ensure the service principal has all the required permissions to access content in your Azure key vault instance. If not, you can run the following using the Azure CLI:

Assign Reader Role to the service principal for your keyvault
```
az role assignment create --role Reader --assignee <aadClientId> --scope /subscriptions/<subscriptionid>/resourcegroups/<resourcegroup>/providers/Microsoft.KeyVault/vaults/<keyvaultname>

az keyvault set-policy -n $KV_NAME --key-permissions get --spn <YOUR SPN CLIENT ID>
az keyvault set-policy -n $KV_NAME --secret-permissions get --spn <YOUR SPN CLIENT ID>
az keyvault set-policy -n $KV_NAME --certificate-permissions get --spn <YOUR SPN CLIENT ID>
```
**Release note**:
```
add key vault support in static provisioning scenario
```

### CODE REVIEW

File: pkg/blobfuse/azure.go [hunk 1]
```diff
@@ -20,8 +20,15 @@ import (
 	"fmt"
 	"os"
 
+	kv "github.com/Azure/azure-sdk-for-go/services/keyvault/2016-10-01/keyvault"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	aazure "github.com/Azure/go-autorest/autorest/azure"
```
Comment (andyzhangx) [inline pkg/blobfuse/azure.go]: `aazure` ?

File: pkg/blobfuse/blobfuse.go [hunk 2]
```diff
@@ -228,3 +230,88 @@ func checkContainerNameBeginAndEnd(containerName string) bool {
 
 	return false
 }
+
+// checkSASToken checks if the key contains the patterns. Because a SAS Token must have these strings, use them to judge.
+func checkSASToken(key string) bool {
```
Comment (andyzhangx) [inline pkg/blobfuse/blobfuse.go]: isSASToken

File: pkg/blobfuse/blobfuse_test.go [hunk 3]
```diff
@@ -322,3 +322,50 @@ func TestCheckContainerNameBeginAndEnd(t *testing.T) {
 		}
 	}
 }
+
+func TestCheckSASToken(t *testing.T) {
+	tests := []struct {
+		key      string
+		expected bool
+	}{
+		{
```
Comment (andyzhangx) [inline pkg/blobfuse/blobfuse_test.go]: also add a storage account key case

File: pkg/blobfuse/blobfuse.go [hunk 4]
```diff
@@ -228,3 +230,88 @@ func checkContainerNameBeginAndEnd(containerName string) bool {
...
+	patterns := []string{"&ss=", "&srt=", "&sp=", "&se=", "&st=", "&spr=", "&sig="}
+	for _, v := range patterns {
+		if !strings.Contains(key, v) {
+			return false
+		}
+	}
+	return true
+}
+
+func (d *Driver) getStorageAccountAndContainer(ctx context.Context, volumeID string, attrib, secrets map[string]string) (string, string, string, string, error) {
```
Comment (andyzhangx) [inline pkg/blobfuse/blobfuse.go]: show return value in the func comment

File: pkg/blobfuse/blobfuse.go [hunk 5]
```diff
@@ -160,9 +162,9 @@ func appendDefaultMountOptions(mountOptions []string) []string {
 
 // get storage account from secrets map
 // returns <accountName, accountKey, accountSasToken>
-func getStorageAccount(secrets map[string]string) (string, string, string, error) {
+func getStorageAccountFromSecretsMap(secrets map[string]string) (string, string, string, error) {
```
Comment (andyzhangx) [inline pkg/blobfuse/blobfuse.go]: add a unit test for getStorageAccountFromSecretsMap

File: pkg/blobfuse/azure.go (hunk 1 above)
Comment (ZeroMagic) [reply pkg/blobfuse/azure.go]: done.

File: pkg/blobfuse/blobfuse.go (hunk 2 above)
Comment (ZeroMagic) [reply pkg/blobfuse/blobfuse.go]: done.

File: pkg/blobfuse/blobfuse_test.go (hunk 3 above)
Comment (ZeroMagic) [reply pkg/blobfuse/blobfuse_test.go]: done.

File: pkg/blobfuse/blobfuse.go (hunk 4 above)
Comment (ZeroMagic) [reply pkg/blobfuse/blobfuse.go]: done.

File: pkg/blobfuse/blobfuse.go (hunk 5 above)
Comment (ZeroMagic) [reply pkg/blobfuse/blobfuse.go]: done.

**Pull request merged**